vm/src/stdlib/collections.rs

```diff
@@ -382,8 +382,8 @@ mod _collections {
         #[pymethod(magic)]
         fn repr(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<String> {
             let repr = if let Some(_guard) = ReprGuard::enter(vm, zelf.as_object()) {
-                let elements = zelf
-                    .borrow_deque()
+                let deque = zelf.borrow_deque().clone();
```

Comment:
Weirdly enough, this can't be

```rust
let elements = zelf.borrow_deque().clone()
    .iter()
    .map(|obj| vm.to_repr(obj))
    .collect::<Result<Vec<_>, _>>()?;
```

It _seems_ as if rust compiles this down to a loop that goes through the elements, clones them and calls map on them  one by one. Moving it on a separate statement avoids this.

Reply:
isn't it because of the lifecycle of borrow?